File: num2words.java
import java.util.Scanner;
import java.util.ArrayList;
import java.lang.Math;

public class Main {
    public static void main(String[] args) {
        Scanner scanner=new Scanner(System.in);

        System.out.print("Enter your number to be modified:");
        int e=scanner.nextInt();
        scanner.close();
        String[] a={"ONE","TWO","THREE","FOUR","FIVE","SIX","SEVEN","EIGHT","NINE","TEN","ELEVEN","TWELVE","THIRTEEN","FOURTEEN","FIFTEEN","SIXTEEN","SEVENTEEN","EIGHTEEN","NINETEEN"};
        String[] b={"TWENTY","THIRTY","FORTY","FIFTY","SIXTY","SEVENTY","EIGHTY","NINETY"};
        String[] c={"HUNDRED","THOUSAND","LAKH","CRORE"};
        ArrayList<String> d=new ArrayList<>();

        //=0
        if(e==0) {
            System.out.print("ZERO");
        }
        
        //<0
        if(e<0) {
            e=-e;
        }

    //Crore
    int f=Math.floorDiv(e%(1000000000), 10000000);
    if (f<20 && f>0) {
        d.add(a[f-1]);
        d.add(c[3]);
    }
    else if (f>19) {
        if (f%10==0) {
            d.add(b[Math.floorDiv(f, 10)-2]);
            d.add(c[3]);
        }
        else if (f%10>0) {
            d.add(b[Math.floorDiv(f, 10)-2]);
            d.add(a[-1+f%10]);
            d.add(c[3]);
        }
    }

    //Lakh
    f=Math.floorDiv(e%10000000, 100000);
    if (f<20 && f>0) {
        d.add(a[f-1]);
        d.add(c[2]);
    }
    else if (f>19) {
        if (f%10==0) {
            d.add(b[Math.floorDiv(f, 10)-2]);
            d.add(c[2]);
        }
        else if (f%10>0) {
            d.add(b[Math.floorDiv(f, 10)-2]);
            d.add(a[-1+f%10]);
            d.add(c[2]);
        }
    }

    //Thousand
    f=Math.floorDiv(e%100000, 1000);
    if (f<20 && f>0) {
        d.add(a[f-1]);
        d.add(c[1]);
    }
    else if (f>19) {
        if (f%10==0) {
            d.add(b[Math.floorDiv(f, 10)-2]);
            d.add(c[1]);
        }
        else if (f%10>0) {
            d.add(b[Math.floorDiv(f, 10)-2]);
            d.add(a[-1+f%10]);
            d.add(c[1]);
        }
    }

    //Hundred
    if (Math.floorDiv(e%1000, 100)>0) {
        d.add(a[-1+Math.floorDiv(e%1000, 100)]);
        d.add(c[0]);
    }

    //<100
    f=e%100;
    if (f<20 && f>0) {
        d.add(a[f-1]);
    }
    else if (f>19) {
        if (f%10==0) {
            d.add(b[Math.floorDiv(f, 10)-2]);
        }
        else if (f%10>0) {
            d.add(b[Math.floorDiv(f, 10)-2]);
            d.add(a[-1+f%10]);
        }
    }

    for(String element : d) {
        System.out.print(element+" ");
    }
    System.out.println();
    }
}

File: num2words.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>
#include<string.h>

int main() {
    char *a[]={"ONE","TWO","THREE","FOUR","FIVE","SIX","SEVEN","EIGHT","NINE","TEN","ELEVEN","TWELVE","THIRTEEN","FOURTEEN","FIFTEEN","SIXTEEN","SEVENTEEN","EIGHTEEN","NINETEEN"};
    char *b[]={"TWENTY","THIRTY","FORTY","FIFTY","SIXTY","SEVENTY","EIGHTY","NINETY"};
    char *c[]={"HUNDRED","THOUSAND","LAKH","CRORE"};
    int e,f,flag=0;
    scanf("%d",&e);
    char d[10][10];

    //Crore
    f=(e%1000000000)/(10000000);
    if (f<20 && f>0) {
        strcpy(d[flag++],a[f-1]);
        strcpy(d[flag++],c[3]);
    }
    else if (f>19) {
        if (f%10==0) {
            strcpy(d[flag++],b[(f/10)-2]);
            strcpy(d[flag++],c[3]);
        }
        else if(f%10>0) {
            strcpy(d[flag++],b[(f/10)-2]);
            strcpy(d[flag++],a[-1+f%10]);
            strcpy(d[flag++],c[3]);
        }
    }

    //Lakh
    f=(e%10000000)/(100000);
    if (f<20 && f>0) {
        strcpy(d[flag++],a[f-1]);
        strcpy(d[flag++],c[2]);
    }
    else if (f>19) {
        if (f%10==0) {
            strcpy(d[flag++],b[(f/10)-2]);
            strcpy(d[flag++],c[2]);
        }
        else if (f%10>0) {
            strcpy(d[flag++],b[(f/10)-2]);
            strcpy(d[flag++],a[-1+f%10]);
            strcpy(d[flag++],c[2]);
        }
    }

    //Thousand
    f=(e%100000)/1000;
    if (f<20 && f>0) {
        strcpy(d[flag++],a[f-1]);
        strcpy(d[flag++],c[1]);
    }
    else if (f>19) {
        if (f%10==0) {
            strcpy(d[flag++],b[(f/10)-2]);
            strcpy(d[flag++],c[1]);
        }
        else if (f%10>0) {
            strcpy(d[flag++],b[(f/10)-2]);
            strcpy(d[flag++],a[-1+f%10]);
            strcpy(d[flag++],c[1]);
        }
    }

    //Hundred
    if ((e%1000)/100>0) {
        strcpy(d[flag++],a[-1+(e%1000)/100]);
        strcpy(d[flag++],c[0]);
    }

    //<100
    f=e%100;
    if (f<20 && f>0) {
        strcpy(d[flag++],a[f-1]);
    }
    else if (f>19) {
        if (f%10==0) {
            strcpy(d[flag++],b[(f/10)-2]);
        }
        else if (f%10>0) {
            strcpy(d[flag++],b[(f/10)-2]);
            strcpy(d[flag++],a[-1+f%10]);
        }
    }
    

    //Display
    for(int i=0;i<flag;i++) {
        printf("%s ",d[i]);
    }
}